### PR TITLE
Remove linkcheck parser from docs pages

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/a1E2H77k/608-update-contributing-docs-to-reflect-latest-docs-build-changes - when Sphinx was upgraded, the jobs were reconfigured and no longer use the linkcheck parser so they now fail (red) when there are broken links and you need to look in the console output to find the issue.

### CODE REVIEW

File: contributing/ci-docs.rst [hunk 1]
```diff
@@ -363,38 +363,9 @@ All are currently built from the master branches despite the build names.
...
-Priority FAILED UNSTABLE
-======== ====== ========
-High	 0
-Normal			0
-Low				10
-======== ====== ========
+Linkchecking
+^^^^^^^^^^^^
+
+Use of the linkcheck output parser was discontinued when the version of Sphinx
```
Comment (sbesson) [inline contributing/ci-docs.rst]: Reading through it for the first time, the immediate question would be "which linkckeck output parser". 
I am unclear whether this section is required at all or if we should drop if and have this policy change recorded via the Git/GitHub history.

**Review by sbesson — COMMENTED** [review]
One comment about the linkchecking section. Otherwise good to merge and deploy.

File: contributing/ci-docs.rst (hunk 1 above)
Comment (hflynn) [reply contributing/ci-docs.rst]: I initially just dropped it altogether then wondered if I should leave something. I'll delete it again.

**Comment by hflynn** [comment]
Fixed @sbesson comment, merging and deploying

**Pull request merged**